helm/raster/config/nginx.conf

```diff
@@ -1,41 +1,68 @@
+load_module modules/ngx_http_js_module.so;
+load_module /otel/otel_ngx_module.so;
 
-{{- $chartName := include "pycsw.name" . -}}
-{{- $serviceName := include "service.fullname" . -}}
-{{- $releaseName := .Release.Name -}}
 
-map $msec $nanosec {
-    "~*(\d{10})\.(\d{3})" "$1$2000000";
-}
 
-upstream pycsw {
-    server {{ $serviceName }}:{{ .Values.servicePort }};
+events {
+    worker_connections  1024;
 }
 
-js_import main from "/etc/nginx/jwt.js";
-js_set $jwt_payload_sub main.jwt_payload_sub;
-
-server {
-    listen      {{ .Values.nginx.targetPort }};
-    # the domain name it will serve for
-    server_name pycsw;
-    # max upload size, adjust to taste
-    keepalive_timeout  500;
-    proxy_connect_timeout       600;
-    proxy_send_timeout          600;
-    send_timeout                600;
-    client_max_body_size        5000;
-    client_header_timeout       600;
-    client_body_timeout         600;
-    client_header_buffer_size   12288; # 12K
-    large_client_header_buffers 4 12288; # 12K
-    fastcgi_read_timeout        300;
-    location /liveness {
-        return 200 "I'm alive\n";
-    }
-    location / {
-        include uwsgi_params;
-        uwsgi_pass pycsw;
-        uwsgi_hide_header    Set-Cookie;  # ensures the header will not be passed back to the client
-        uwsgi_param HTTP_Cookie ""; # ensures that a client cannot pass any prior cookies to the webapp and spoil your cache
-    }
+
+http {
+    opentelemetry_config /otel/otel-nginx.toml;
+    include       /etc/nginx/mime.types;
+    default_type  application/octet-stream;
+
+    log_format json escape=json 
+    '{'
+        '"Timestamp": $nanosec,'
+        '"Attributes": {'
+            '"time_local": "$time_local",'
+            '"http.status_code": $status,'
+            '"http.url": "$http_referer",'
+            '"http.x_forwarded_for": "$http_x_forwarded_for",'
+            '"http.user_agent": "$http_user_agent",'
+            '"http.request_time": $request_time,'
+            '"request_length": "$request_length",'
+            '"http.host": "$host",'
+            '"upstream_response_time": $upstream_response_time,'
+            '"body_bytes_sent": "$body_bytes_sent",'
+            '"remote_addr": "$remote_addr",'
+            '"remote_user": "$remote_user",'
+            '"connection":"$connection",'
+            '"connection_requests": "$connection_requests",'
+            '"connections_active": "$connections_active",'
+            '"connections_reading": "$connections_reading",'
+            '"connections_writing": "$connections_writing",'
+            '"connections_waiting": "$connections_waiting"'
+        '},'
+        #'"TraceId": "$opentelemetry_trace_id",' ## this is a byte sequence (hex-encoded in JSON)
+        #'"SpanId": "$opentelemetry_span_id",'
+        '"TokenUser": "$jwt_payload_sub",'
+        '"SeverityText": "INFO",'
+        '"SeverityNumber": 9,'
+        '"RequestBody": "$request_body",'
+        '"Body": "$request"'
+    '}';
```

Comment:
Log format should use MapColonies opentelemetry convention:
https://mapcolonies.atlassian.net/wiki/spaces/MAPConflicResolution/pages/1876328458/MapColonies+OpenTelemetry+Convention